### DISCUSSION TITLE
Naming suggestion: maybe do readXxx() and writeXxx() instead of xxx() and a single write()

### DISCUSSION THREAD

**Discussion by iliazeus** [humanwhocodes/humanfs#15] [Feedback]
- `.readText()`, `.readJson()`, `.readArrayBuffer()` instead of `.text()`, `.json()`, `.arrayBuffer()`
- `.writeText()`, `.writeJson()`, `.writeArrayBuffer()` instead of a single `.write()` for all

pros:
- it allows for a nice paring of input and output functions, increasing discoverability
- it allows for having a nice unambiguous interface for writing JSON files, without a manual `JSON.stringify()`
- it's up to your coding style ofc, but verb-based names seem to be more natural for this case

cons:
- a couple letters longer i guess?

**Comment by nzakas** [comment]
Thanks for the feedback. I'm sticking with `text()`, `json()`, etc. because these mirror what is available with `fetch()`. I don't mind needing to use `JSON.stringify()` with `write()`, because it's hard to know how people would want their JSON formatted in the output, and I'm trying to avoid having a bunch of optional parameters for methods.
